packages/drivers/odsp-driver/src/odspUtils.ts

```diff
@@ -239,6 +243,14 @@ export interface INewFileInfo {
     createLinkType?: ShareLinkTypes | ISharingLinkKind;
 }
 
+export interface IExistingFileInfo extends IFileInfoBase {
+    itemId: string;
+}
+
+export function isNewFileInfo(fileInfo: INewFileInfo | IExistingFileInfo): fileInfo is INewFileInfo {
```

Comment:
this should be a discriminated union: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions

Reply:
Thanks, updated to discriminated union